### PR TITLE
Removes statsd output

### LINKED ISSUE #2982: Future of StatsD outputs
The k6-core team recently had an internal meeting to discuss the current status of StatsD outputs in k6, the several issues reported by users, and the best way to move forward that benefits both users, and maintenance of k6.

## Background

The [`statsd` output](https://k6.io/docs/results-output/real-time/statsd/) is a generic implementation of the [StatsD line protocol](https://github.com/statsd/statsd#usage). It allows pushing metrics to any StatsD server, and, in theory, any StatsD backend.

Under the hood, it uses the [DogStatsD client library](https://github.com/DataDog/datadog-go), but it doesn't use any Datadog-specific functionality. This is why the same output can be used to send metrics to both [Datadog](https://k6.io/docs/results-output/real-time/datadog/) and [New Relic](https://k6.io/docs/results-output/real-time/new-relic/).

## Issues

The problem is that, on its own, this generic implementation is very basic, and only supports [3 metric types](https://github.com/statsd/statsd/blob/master/docs/metric_types.md) (Counters, Gauges, and Sets). Datadog and New Relic have extended the protocol to add support for more metric types, though these changes, are not standard, and likely wouldn't work across different servers and backends. So we can't use this functionality in the `statsd` output without risking breaking generic StatsD support, or specifically supporting individual backends, which we want to avoid.

There are several issues reported by users on the forum:

- https://community.k6.io/t/trend-metrics-on-datadog-percentile/3995
- https://community.k6.io/t/http-req-duration-wrong-datadog/440
- https://community.k6.io/t/why-do-k6-http-req-duration-max-and-k6-http-req-duration-95percentile-have-the-same-value-in-datadog/5487
- https://community.k6.io/t/percentiles-calculation-on-new-relic/3690
- https://community.k6.io/t/how-to-get-the-correct-values-for-p-90-on-k6-newrelic/5192

Some of these, particularly those related to percentile values, and summarized in #2819, could be fixed by using the non-standard metric types, such as [Distribution](https://docs.datadoghq.com/metrics/distributions/). But, as mentioned above, we can't use it directly in the output.

## Proposed solutions

We agreed that the way forward consists of several steps:

1. Research the status of alternative protocol support for Datadog and New Relic.

   For example, [New Relic supports Prometheus Remote Write natively](https://docs.newrelic.com/docs/infrastructure/prometheus-integrations/get-started/send-prometheus-metric-data-new-relic/#remote-write), which k6 can output via [xk6-output-prometheus-remote](https://github.com/grafana/xk6-output-prometheus-remote).

   Datadog only supports [pulling metrics from Prometheus endpoints](https://docs.datadoghq.com/containers/kubernetes/prometheus/) (not Remote Write), but could technically support it via [Vector](https://github.com/vectordotdev/vector) (owned by Datadog), which does have a [RW source](https://vector.dev/docs/reference/configuration/sources/prometheus_remote_write/).

2. Depending on the results of the above research, create separate k6 Datadog and New Relic output extensions. These will initially not solve any of the reported issues, and will essentially be an extraction and duplication of the current `statsd` output. The benefit of this is that being separate outputs would allow them to use any backend-specific functionality.

   The k6-core team will be in charge of this development and initial maintenance.

   The reason this depends on the research from step 1, is because if we find that both backends work fine with Prometheus RW, then we can just update our documentation instead of creating separate extensions. Or if only New Relic does, then only create the extension for Datadog, etc.

3. Contact the respective Datadog and New Relic teams, explain our situation and existing issues, and politely transfer maintenance of the extensions to them.

   We believe their teams will have better knowledge about the best ways to fix the current issues, and would be better equipped to continue maintaining the integration with their backends.

   We would essentially like to achieve the same situation as with the current [Elasticsearch output](https://github.com/elastic/xk6-output-elasticsearch), which is maintained by Elastic.

4. Decide whether to continue supporting the plain StatsD output in k6, or if it should be deprecated in favor of Prometheus or [OpenTelemetry](https://github.com/grafana/k6/issues/2557).

   [The StatsD project](https://github.com/statsd/statsd) itself is quite old, and hasn't seen updates in years. With all its limitations, it's unlikely anyone is still using it. Especially since nowadays most backends support modern protocols, like Prometheus, OpenTelemetry or OpenMetrics.

I can create sub-issues for each step if it makes more sense. Let me know.

### PR DESCRIPTION
**Pull request by olegbespalov**
## What?

Removes StatsD output from k6 in favor of https://github.com/LeonAdato/xk6-output-statsd

## Why?

An external extension exists https://github.com/LeonAdato/xk6-output-statsd.

## Checklist

- [ ] I have performed a self-review of my code.
- [ ] I have added tests for my changes.
- [ ] I have run linter locally (`make lint`) and all checks pass.
- [ ] I have run tests locally (`make tests`) and all tests pass.
- [ ] I have commented on my code, particularly in hard-to-understand areas.

## Related PR(s)/Issue(s)

https://github.com/grafana/k6/pull/3848

Closes #2982

### CODE REVIEW

**Review by mstoykov — DISMISSED** [review]
LGTM

But am not certain what is going on with windows packaging

**Comment by joanlopez** [comment]
> But am not certain what is going on with windows packaging

Looks like it was a temporary error, something wrong with the NuGet backend made it impossible to download & install one of the dependencies we use for packaging:

```sh
[NuGet] Response status code does not indicate success: 503 (Service Unavailable: Back-end server is at capacity).
```

I just clicked on retry, and it worked now. Apparently they managed whatever was wrong with their backend.

**Review by joanlopez — APPROVED** [review]
Except for the tiny typo, it looks good! 🎉

File: cmd/outputs.go [hunk 1]
```diff
@@ -52,12 +51,11 @@ func getAllOutputConstructors() (map[string]output.Constructor, error) {
 			return nil, errors.New("the kafka output was deprecated in k6 v0.32.0 and removed in k6 v0.34.0, " +
 				"please use the new xk6 kafka output extension instead - https://github.com/k6io/xk6-output-kafka")
 		},
-		builtinOutputStatsd.String(): func(params output.Params) (output.Output, error) {
-			params.Logger.Warn("The statsd output is deprecated, and will be removed in k6 v0.55.0 " +
-				"Please use the new xk6 statsd output extension instead. " +
+		builtinOutputStatsd.String(): func(_ output.Params) (output.Output, error) {
+			return nil, errors.New("the statsd output was deprecated in k6 v0.47.0 and removed in in k6 v0.55.0, " +
+				"please use the new xk6 statsd output extension instead. " +
```
[the remark this answers is not in the document]
Comment (olegbespalov) [reply cmd/outputs.go]: Good catch! Thanks :+1:

**Review by joanlopez — APPROVED** [review]

**Review by mstoykov — APPROVED** [review]

**Pull request merged**